src/analyzer/analyzerwaveform.cpp

```diff
@@ -75,6 +82,11 @@ bool AnalyzerWaveform::initialize(TrackPointer tio, int sampleRate, int totalSam
         m_waveformSummary = WaveformPointer(new Waveform(
                 sampleRate, totalSamples, mainWaveformSampleRate,
                 summaryWaveformSamples));
+        //I force it to NOT dirty, because other Mixxx threads like the Main thread can trigger 
+        //TrackDAO.updateTrack(), which would try to save an empty or incomplete analysis. 
+        m_waveform->setDirty(false);
```

Comment:
Ah thank you for explanations. It looks like a dirty flag alone is ambiguous.
How about turn it into a state variable: 
0 = Virgin 
1 = Changing 
2 = SavePending 
3 = Saved